server.js

```diff
@@ -0,0 +1,92 @@
+import express from 'express'
+import Student from  './studentSchema.js'
+import morgan from 'morgan'
+import cors from 'cors'
+import './db-connect.js'
+
+// establish connection to database
+
+const app = express()
+
+
+//middlewares
+app.use( express.json())
+app.use(cors())
+app.use(morgan("dev"))
+
+// routes 
+
+// get all students
+app.get("/students", async(req, res) =>{
+  const students = await Student.find()
+  console.log(students)
+  res.json(students)
+})
+
+// get a student
+app.get("/students/:id", async(req, res, next) => {
+   
+    try {
+        const {id} = req.params
+        const student = await Student.findById(id)
+        res.json(student)
+    } catch(err){
+        next(err)
+    }
+})
+
+// create
+app.post("/students", async(req, res, next) =>{
+    
+    try{
+        const body = req.body
+        const newStudent = await Student.create(body)
+        res.json(newStudent)
+    } catch(err){
+        next(err)
+    }
+})
+
+// delete
+app.delete("/students/:id", async(req, res, next) => {
+    try{
+        const {id} = req.params
+        const student = await Student.findByIdAndDelete(id)
+        res.json(student)
+    } catch(err){
+        next(err)
+    }
+    
+})
+
+// update
+
+app.put("/students/:id", async(req, res) => {
+    try{ const {id} = req.params
+    const studentData = req.body
+    const studentUpdated = await Student.findByIdAndUpdate(id, studentData)
```

Comment:
This is fine, but findByIdAndUpdate would now return you the OLD state of the student object.

To return the updated one from the database, you can append here the `{new: true}` option as third parameter.